cylc/flow/data_store_mgr.py

```diff
@@ -2078,62 +2088,30 @@ def _family_ascent_point_update(self, fp_id):
                 tp_delta = tp_updated.get(tp_id)
                 tp_node = tp_added.get(tp_id, tp_data.get(tp_id))
 
-                tp_state = tp_delta
-                if tp_state is None or not tp_state.HasField('state'):
-                    tp_state = tp_node
-                if tp_state.state:
-                    task_states.append(tp_state.state)
-
-                tp_held = tp_delta
-                if tp_held is None or not tp_held.HasField('is_held'):
-                    tp_held = tp_node
-                if tp_held.is_held:
-                    is_held_total += 1
-
-                tp_queued = tp_delta
-                if tp_queued is None or not tp_queued.HasField('is_queued'):
-                    tp_queued = tp_node
-                if tp_queued.is_queued:
-                    is_queued_total += 1
-
-                tp_runahead = tp_delta
-                if (
-                    tp_runahead is None
-                    or not tp_runahead.HasField('is_runahead')
-                ):
-                    tp_runahead = tp_node
-                if tp_runahead.is_runahead:
-                    is_runahead_total += 1
-
-                tp_retry = tp_delta
-                if tp_retry is None or not tp_retry.HasField('is_retry'):
-                    tp_retry = tp_node
-                if tp_retry.is_retry:
-                    is_retry = True
-
-                tp_wallclock = tp_delta
-                if (
-                    tp_wallclock is None
-                    or not tp_wallclock.HasField('is_wallclock')
-                ):
-                    tp_wallclock = tp_node
-                if tp_wallclock.is_wallclock:
-                    is_wallclock = True
-
-                tp_xtriggered = tp_delta
-                if (
-                    tp_xtriggered is None
-                    or not tp_xtriggered.HasField('is_xtriggered')
-                ):
-                    tp_xtriggered = tp_node
-                if tp_xtriggered.is_xtriggered:
-                    is_xtriggered = True
-
-                tp_depth = tp_delta
-                if tp_depth is None or not tp_depth.HasField('graph_depth'):
-                    tp_depth = tp_node
-                if tp_depth.graph_depth < graph_depth:
-                    graph_depth = tp_depth.graph_depth
+                # lovely future case for walrus operator:
+                tp_state = self._helper(tp_delta, tp_node, 'state')
+                if tp_state:
+                    task_states.append(tp_state)
+
+                for label, counter in [
+                    ("is_held", is_held_total),
+                    ("is_queued", is_queued_total),
+                    ("is_runahead", is_runahead_total),
+                ]:
+                    if self._helper(tp_delta, tp_node, label):
+                        counter += 1
```

Comment:
I think this increments a local variable called `counter` rather than changing the actual counter (e.g. `is_held_total`).